tests/Graph/TraverserCacheTest.cpp

```diff
@@ -0,0 +1,205 @@
+////////////////////////////////////////////////////////////////////////////////
+/// DISCLAIMER
+///
+/// Copyright 2014-2021 ArangoDB GmbH, Cologne, Germany
+/// Copyright 2004-2014 triAGENS GmbH, Cologne, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Heiko Kernbach
+////////////////////////////////////////////////////////////////////////////////
+
+#include "gtest/gtest.h"
+
+#include "fakeit.hpp"
+
+#include "Aql/AqlValue.h"
+#include "Aql/Query.h"
+#include "Aql/QueryWarnings.h"
+#include "Cluster/ServerState.h"
+#include "Graph/Cache/RefactoredTraverserCache.h"
+//#include "Graph/ClusterTraverserCache.h"
+#include "Graph/GraphTestTools.h"
+#include "Graph/TraverserOptions.h"
+#include "Transaction/Methods.h"
+
+#include <Aql/TraversalStats.h>
+#include <velocypack/Builder.h>
+#include <velocypack/Slice.h>
+#include <velocypack/velocypack-aliases.h>
+
+using namespace arangodb;
+using namespace arangodb::aql;
+using namespace arangodb::graph;
+
+namespace arangodb {
+namespace tests {
+namespace traverser_cache_test {
+
+class TraverserCacheTest : public ::testing::Test {
+ protected:
+  ServerState::RoleEnum oldRole;
+
+  graph::GraphTestSetup s{};
+  graph::MockGraphDatabase gdb;
+  arangodb::aql::TraversalStats stats{};
```

Comment:
I am missing all asserts on the stats.
Please add them (e.g. after reading a vertex, it should have increased "read index"

Reply:
Done.